pallets/handles/src/tests/handle_retirements_tests.rs

```diff
@@ -53,6 +54,64 @@ fn claim_and_retire_handle_happy_path() {
 	});
 }
 
+#[test]
+fn test_handle_early_retirement_fails() {
+	new_test_ext().execute_with(|| {
+		let base_handle_str = "test1";
+
+		let alice = sr25519::Pair::from_seed(&[0; 32]);
+		let expiration = 50;
+		let (payload, proof) =
+			get_signed_claims_payload(&alice, base_handle_str.as_bytes().to_vec(), expiration);
+		assert_ok!(Handles::claim_handle(
+			RuntimeOrigin::signed(alice.public().into()),
+			alice.public().into(),
+			proof.clone(),
+			payload.clone()
+		));
+		let msa_id = MessageSourceId::decode(&mut &alice.public().encode()[..]).unwrap();
+		let handle = Handles::get_handle_for_msa(msa_id);
+		assert!(handle.is_some());
+
+		// Compose the full display handle from the base handle, "." delimeter and suffix
+		let handle_result = handle.unwrap();
```

Comment:
i think there is a function here in handles you can use for this.

Reply:
That is with the suffix index, at this place we are getting handle from state and forming a display handle (ow a consumer of rpc would do)